### PR TITLE
Allow for bugzilla management to use a minimum threshold for 'reopene…

### PR DESCRIPTION
**Pull request by jmaher**
…d bugs'

### CODE REVIEW

**Comment by jmaher** [comment]
the goal here is to allow for a configurable minimum- right now it is 1 and this means when we get at least 1 new failure for a given bug we reopen the bug.  If we set the minimum to 3, then on the 3rd bug we would reopen the bug.  This is configured and tested properly.  If we wanted to use other criteria for selecting the minimum, possibly we need to solve this in another location in the code.

**Comment by jmaher** [comment]
> ## [Codecov](https://app.codecov.io/gh/mozilla/treeherder/pull/8144?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla) Report
> 
> Attention: Patch coverage is `94.44444%` with `1 line` in your changes missing coverage. Please review.
> 
> > Project coverage is 77.17%. Comparing base [(`f4fcf9a`)](https://app.codecov.io/gh/mozilla/treeherder/commit/f4fcf9a8616fa51a1781ad515752381887f01a68?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla) to head [(`b8a856f`)](https://app.codecov.io/gh/mozilla/treeherder/commit/b8a856f90e8948a3916ea8162b9f96d51370381b?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla).
> > Report is 7 commits behind head on master.
> 
> [Files](https://app.codecov.io/gh/mozilla/treeherder/pull/8144?dropdown=coverage&src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla) 	Patch % 	Lines
> [tests/etl/test_bugzilla.py](https://app.codecov.io/gh/mozilla/treeherder/pull/8144?src=pr&el=tree&filepath=tests%2Fetl%2Ftest_bugzilla.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla#diff-dGVzdHMvZXRsL3Rlc3RfYnVnemlsbGEucHk=) 	90.90% 	[1 Missing ⚠️ ](https://app.codecov.io/gh/mozilla/treeherder/pull/8144?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla)
> Additional details and impacted files
> 
> [☔ View full report in Codecov by Sentry](https://app.codecov.io/gh/mozilla/treeherder/pull/8144?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla). 📢 Have feedback on the report? [Share it here](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=mozilla).

odd, this codecov report indicates missing coverage although I have made the test case more robust.  it doesn't produce a lot of verbose information in the test run in CI, but locally it ran through the extra test case (via parameterize)

**Comment by jmaher** [comment]
please note, this doesn't work for NEW bugs, only for adjusting when a bug is "re-opened".  

if this approach in this PR is ok, I would like to merge it and build upon it to support NEW bugs.

File: tests/etl/test_bugzilla.py [hunk 1]
```diff
@@ -53,19 +74,21 @@ def test_bz_reopen_bugs(request, mock_bugzilla_reopen_request, client, test_job,
             {
                 "status": "REOPENED",
                 "comment": {
-                    "body": "New failure instance: https://treeherder.mozilla.org/logviewer?job_id=1&repo=mozilla-central"
+                    "body": "New failure instance: https://treeherder.mozilla.org/logviewer?job_id=5&repo=mozilla-central"
```
Comment (Archaeopteryx) [inline tests/etl/test_bugzilla.py:77]: Understanding the returned test data based on the code generation took some time.

**Review by Archaeopteryx — APPROVED** [review]

**Pull request merged**